pkl-core/src/main/java/org/pkl/core/http/HttpClient.java

```diff
@@ -108,6 +108,15 @@ interface Builder {
      */
     Builder addBuiltInCertificates();
 
+    /**
+     * Sets a test server's listening port.
+     *
+     * <p>If set, requests that specify port 12110 are rewritten to use the given port. This enables
+     * users of this client to connect to a test server without having to know its (ephemeral)
+     * listening port.
```

Comment:
```suggestion
     * listening port.
     *
     * <p>This is an internal option used for testing purposes only.
```

Reply:
I didn't see a need to "hide" this option at the `HttpClient` level. Anyone who codes against `HttpClient`, rather than just passing it around, may find this option useful.

Should switch from 12210 to 0 in a follow-up PR.

Reply:
It's the same with `org.pkl.commons.cli.CliBaseOptions`; both are public APIs but expose flags that are only meant for test purposes (e.g. `testMode`, `testPort` added by this PR) and we call them "internal" in the doc comments.